errors/abci_test.go

```diff
@@ -156,34 +159,79 @@ func TestABCIInfoHidesStacktrace(t *testing.T) {
 }
 
 func TestRedact(t *testing.T) {
-	if err := Redact(ErrPanic, false); ErrPanic.Is(err) {
-		t.Error("in non-debug mode, redact must not pass through panic error")
+	if err := Redact(ErrPanic); ErrPanic.Is(err) {
+		t.Error("reduct must not pass through panic error")
 	}
-	if err := Redact(ErrPanic, true); !ErrPanic.Is(err) {
-		t.Error("in debug mode, redact should pass through panic error")
-	}
-
-	if err := Redact(ErrNotFound, true); !ErrNotFound.Is(err) {
-		t.Error("in debug mode, redact should pass through weave error")
-	}
-	if err := Redact(ErrNotFound, false); !ErrNotFound.Is(err) {
-		t.Error("in non-debug mode, redact should pass through weave error")
+	if err := Redact(ErrNotFound); !ErrNotFound.Is(err) {
+		t.Error("reduct should pass through weave error")
 	}
 
 	var cerr customErr
-	if err := Redact(cerr, true); err != cerr {
-		t.Error("in debug mode, redact should pass through ABCI code error")
-	}
-	if err := Redact(cerr, false); err != cerr {
-		t.Error("in non-debug mode, redact should pass through ABCI code error")
+	if err := Redact(cerr); err != cerr {
+		t.Error("reduct should pass through ABCI code error")
 	}
 
 	serr := fmt.Errorf("stdlib error")
-	if err := Redact(serr, false); err == serr {
-		t.Error("in non-debug mode, redact must not pass through a stdlib error")
+	if err := Redact(serr); err == serr {
+		t.Error("reduct must not pass through a stdlib error")
+	}
+}
+
+func TestABCIInfoSerializeErr(t *testing.T) {
+	var (
+		// create errors with stacktrace for equal comparision
+		myErrState          = errors.WithStack(ErrState)
+		myErrMsg            = errors.WithStack(ErrMsg)
+		myPanic             = errors.WithStack(ErrPanic)
+		myErrStateSTJson, _ = json.Marshal(fmt.Sprintf("%+v", myErrState))
+		myErrMsgSTJson, _   = json.Marshal(fmt.Sprintf("%+v", myErrMsg))
+	)
+
+	specs := map[string]struct {
```

Comment:
Nice test cases. Just the comment of using Wrap above, but I think you cover all cases.

It would be nice if we started integrating `cover` in our workflow somehow to see which lines of the PR diff were tested. In some integrated way. Currently the coverage metrics are quite separate and hard to map to the diff

Reply:
Package coverage is quite well: `github.com/iov-one/weave/errors	0.069s	coverage: 84.7% of statements` 🤞

Reply:
:100: :medal_sports: 

When I started weave, I used cover graphs a lot to find untested code paths, as I was writing much code faster than tests, and those graphs really helped me zoom into large chunks that needed to be covered. It's not just the numbers (which are good, no complaint), but also the use of graphs to direct where to test. (This is a general workflow comment... it's been ages for me as well and I kind of forget how I did this as well)